### DISCUSSION TITLE
Propagation loss of guided mode

### DISCUSSION THREAD

**Discussion by berzoll** [NanoComp/meep#2822] [Q&A]
Hello,

I am trying to simulate the propagation loss of a guided mode within a lossy waveguide.
I am struggling to match two definitions I found in meep's documentation, and I am not entirely sure whether these two quantities depict the same thing or not.

**Method#1**: I simulate the band diagram of a unit cell, then I get the complex effective index as k/w. From this I derive the imaginary part of the propagation constant as imag(n)*2*pi/lambda. This gives me a results in accordance with some external mode solver.

**Method#2** (as shown [here](https://meep.readthedocs.io/en/latest/FAQ/#how-do-i-compute-the-effective-index-of-an-eigenmode-of-a-lossy-waveguide)): I simulate the band diagram of a unit cell, calculate the group veloctiy, and then extract the loss per unit length (I believe this should the imaginary part of the propagation constant again) as imag(w)/abs(v_group). I verified the v_group value is correct.

Now, these two quantities are the same if rescaled by a factor of about 3/2*pi (method#1 = method#2 *3/2 *pi, empirically found). I do not understand where this factor would originate from, and whereas I am fully understanding the quantities definitions in the two methods and what they are meant to capture.

**Comment by shahzaib3120** [comment]
@berzoll Can you please provide the code for computing propagation loss or the complex effective index?
Thank you.
